reference/docs-conceptual/learn/ps101/06-flow-control.md

```diff
@@ -0,0 +1,336 @@
+---
+title: Flow control
+description: PowerShell provides methods to create loops, make decisions, and logically control the flow of code in scripts.
+ms.date: 06/02/2020
+ms.topic: guide
+ms.custom: Contributor-mikefrobbins
+ms.reviewer: mirobb
+---
+# Chapter 6 - Flow control
+
+## Scripting
+
+When you move from writing PowerShell one-liners to writing scripts, it sounds a lot more
+complicated than it really is. A script is nothing more than the same or similar commands that you
+would run interactively in the PowerShell console, except they're saved as a `.PS1` file. There are
+some scripting constructs that you may use such as a `foreach` loop instead of the `ForEach-Object`
+cmdlet. To beginners, the differences can be confusing especially when you consider that `foreach`
+is both a scripting construct and an alias for the `ForEach-Object` cmdlet.
+
+## Looping
+
+One of the great things about PowerShell is, once you figure out how to do something for one item,
+it's almost as easy to do the same task for hundreds of items. Simply loop through the items using
+one of the many different types of loops in PowerShell.
+
+### ForEach-Object
+
+`ForEach-Object` is a cmdlet for iterating through items in a pipeline such as with PowerShell
+one-liners. `ForEach-Object` streams the objects through the pipeline.
+
+Although the **Module** parameter of `Get-Command` accepts multiple values that are strings, it only
+accepts them via pipeline input by property name or via parameter input. In the following scenario,
+if I want to pipe two strings by value to `Get-Command` for use with the **Module** parameter, I
+would need to use the `ForEach-Object` cmdlet.
+
+```powershell
+'ActiveDirectory', 'SQLServer' |
+>> ForEach-Object {Get-Command -Module $_} |
+>> Group-Object -Property ModuleName -NoElement |
+>> Sort-Object -Property Count -Descending
+```
+
+```Output
+Count Name
+----- ----
+  147 ActiveDirectory
+   82 SqlServer
+```
+
+In the previous example, `$_` is the current object. Beginning with PowerShell version 3.0,
+`$PSItem` can be used instead of `$_`. But I find that most experienced PowerShell users still
+prefer using `$_` since it's backward compatible and less to type.
+
+When using the `foreach` keyword, you must store all of the items in memory before iterating through
+them, which could be difficult if you don't know how many items you're working with.
+
+```powershell
+$ComputerName = 'DC01', 'WEB01'
+foreach ($Computer in $ComputerName) {
+>>     Get-ADComputer -Identity $Computer
+>> }
+```
```

Comment:
```suggestion
```powershell
$ComputerName = 'DC01', 'WEB01'
foreach ($Computer in $ComputerName) {
  Get-ADComputer -Identity $Computer
}
```
```

Reply:
Fixed